File: Java/src/session3/td/ConsRec.java
package session3.td;

public class ConsRec implements Mot{
	private char caractere;
	private Mot reste;
	private int taille;

	public ConsRec(char i, Mot ens) {
		this.caractere = i;
		this.reste = ens;
		this.taille = 1 + ens.taille();
	}

	@Override
	public boolean casCons() {
		return true;
	}
	
	@Override
	public int taille() {
		return this.taille;
	}

	@Override
	public char caractere() {
		return this.caractere;
	}

	@Override
	public Mot reste() {
		return this.reste;
	}

	/*@Override
	public char caractereA(int index) {
		if(index==0) {
			return this.caractere;
		}else {
			return reste.caractereA(index);
		}
	}*/

	@Override
	public int derniereOccurence(char ch) {
		// Returns the index of the of the character ch

	    if (ch == this.caractere) {
	        // base case: ch is at the beginning of str; return 0
	        return 1; 
	    }

	    // recursive step
	    int subIndex = reste.derniereOccurence(ch);

	    return 1 + subIndex; 
	}
}

File: TD3/typescript/session3/td/ConsRec.ts
import { Mot } from "./Mot";

export class ConsRec extends Mot{
    

    public var_lettre:string;
    public var_reste:Mot;
    public vTaille:number;

    public constructor(i:string, ens:Mot){
        super();
        this.var_lettre=i;
        this.var_reste=ens;
        this.vTaille = 1 + this.var_reste.taille();
    }

    taille(): number {
        return this.vTaille;
    }

    casCons():boolean{
        return true;
    }

    caractere():string{
        return this.var_lettre;
    }

    reste():Mot{
        return this.var_reste;
    }

}